### PR TITLE
Basic Nostr Login with pubkey

### PR DESCRIPTION
**Pull request by davidabram**
- Added nostr-relaypool
- Created Provider and context for nostr-relaypool
- Created useMetadata hook, that gets all events with kind 0 for current user.
- Added an input + button where you can enter your npub, you will be connected to nostr relays and your avatar will be shown.
- Login page done
- Signup page placeholder.
- Shows name of the logged user in the dropdown.
- SignIn SignOut functionality
- saves pubkey in localstorage.
- created useSessuion hook that abstracts useMetadata so you can detect if the user is logged in and what permission levels does they have.

Closes https://github.com/NostrGit/NostrGit/issues/20

### CODE REVIEW

**Comment by brycedev** [comment]
@davidabram this is awesome!

i just wanted to note that we recently setup the [src/app folder](https://beta.nextjs.org/docs/routing/fundamentals), so you'll want to get rid of src/pages/* to get this to build

**Comment by davidabram** [comment]
Can anyone more design-savvy help me out to create a better login design. @brycedev @PeerRich 

Here is the part of the code that [needs some better UI](https://github.com/NostrGit/NostrGit/pull/32/files#diff-89ace09d32ce46210d23a856acf3851bd8a9a3edd7b0706ef308e037ab8ce46bR21)

**Comment by reyamir** [comment]
> Can anyone more design-savvy help me out to create a better login design. @brycedev @PeerRich
> 
> Here is the part of the code that [needs some better UI](https://github.com/NostrGit/NostrGit/pull/32/files#diff-89ace09d32ce46210d23a856acf3851bd8a9a3edd7b0706ef308e037ab8ce46bR21)

I think we should create seperate page for login, instead of put form in header

**Full flow**:
- User input private key as hex string or nsec
- Fetch user metadata
- Save user data (pubkey, privkey and metadata) to local storage

and `NostrProvider` should put at `app/layout.js`, so we can connect to relays at launch and once, then we can query and filter when needed

Example login page:
Imgur

**Comment by PeerRich** [comment]
agree, a new page also means you can send people a signup link:

nostrgit.com/signup

**Comment by davidabram** [comment]
> and `NostrProvider` should put at `app/layout.js`, so we can connect to relays at launch and once, then we can query and filter when needed

Unfortunately you can't use Context.Provider in RSCs, so that's why I have localized to Header for time being, this is still in draft.

File: package.json [hunk 1]
```diff
@@ -3,10 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "build": "next build",
+    "build": "npm run delete:nostr-relaypool-ts && next build",
```
Comment (davidabram) [inline package.json:6]: This is necessary to fix next build.

There is an issue nostr-relaypool; it publishes all the source ts files that don't play nicely with next.js build step.

**Comment by PeerRich** [comment]
excited for this!

step by step

**Comment by cypherhoodlum** [comment]
Fantastic. I too think that the private key should be encrypted using a passphrase or a password. People are used to passwords so the initial "account" creation could start by choosing a username and a password. It should be very clear that the password is only used to encrypt and cannot recover the nsec. I think users should be forced to write down (BIP-39) seed words before proceeding from the signup page.

**Comment by e-roy** [comment]
Love the conversation, I'm very interested in understanding more about security as I'm toying with the idea of building a browser extension.  I don't have much to add here other than a [twitter link](https://twitter.com/tayvano_/status/1630352342473375744?s=20) to conversations on how metamask handles security.  

Hopefully this helps in research around this subject

**Comment by cypherhoodlum** [comment]
No stress, @e-roy! We all do what we can :) Thanks for the link.

**Comment by davidabram** [comment]
@PeerRich I am not able to see why deployment has failed on vercel. Could you copy paste the logs in the comment?

**Comment by PeerRich** [comment]
@davidabram have you tried "yarn build" locally?

**Comment by cypherhoodlum** [comment]
#112 created for future reference.

**Comment by PeerRich** [comment]
@davidabram unborked deployment. works now.

`yarn add ws` was missing

**Comment by PeerRich** [comment]
@b35363 can you re-review? also, maybe your requested changes can be addressed in a follow up PR

**Pull request merged**